stat/roc.go

```diff
@@ -106,7 +106,7 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 
 	invNeg := 1 / nNeg
 	invPos := 1 / nPos
-	for i := range tpr {
+	for i := range tpr[:bin+1] {
```

Comment:
Unless I'm mistaken, this would handle when there is a single "extra" bin above the maximum value in y, but not multiple. I could add a test case for this? I agree it's prettier but I don't think it's equivalent

Reply:
This leaves all elements in `tpr[bin+1:]` and `fpr[bin+1:]` as zero, which AFAICS is the outcome of the existing code in #1561.